### PR TITLE
Add more poly ops needed for lowering polymul

### PR DESCRIPTION
**Pull request by j2kun**
Adding poly.degree, poly.monomial, which will be useful for lowering poly.mul since I can implement the poly remainder routine using poly ops before the lowering starts.

Also encountered a bug in the bit width of the type conversion, and this led me down a long path that ended with a simpler poly.add lowering, thanks to Asra and Shruthi for pair programming on that.

### CODE REVIEW

File: lib/Conversion/PolyToStandard/PolyToStandard.cpp [hunk 1]
```diff
@@ -1,12 +1,15 @@
 #include "include/Conversion/PolyToStandard/PolyToStandard.h"
 
+#include <iostream>
```
Comment (asraa) [inline lib/Conversion/PolyToStandard/PolyToStandard.cpp:3]: nit: remove debugging include

File: tests/poly/lower_monomial.mlir [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#cycl_2048 = #poly.polynomial<1 + x**1024>
+#ring = #poly.ring<cmod=4294967296, ideal=#cycl_2048>
+
+func.func @test_monomial() {
+  // CHECK: arith.constant 1023
+  %deg = arith.constant 1023 : index
+  // CHECK: arith.constant 5
+  %five = arith.constant 5 : i32
+  // CHECK: arith.constant
+  // CHECK: tensor.insert
```
Comment (asraa) [inline tests/poly/lower_monomial.mlir:13]: nit: could you assert that the check inserts at index 5 for correctness?

File: lib/Conversion/PolyToStandard/PolyToStandard.cpp [hunk 3]
```diff
@@ -107,6 +115,71 @@ struct ConvertToTensor : public OpConversionPattern<ToTensorOp> {
...
+        op,
+        /*resultTypes=*/TypeRange{b.getIndexType()},
+        /*operands=*/ValueRange{initIndex.getResult()},
+        /*beforeBuilder=*/
+        [&](OpBuilder &nestedBuilder, Location nestedLoc, ValueRange args) {
+          Value index = args[0];
+          ImplicitLocOpBuilder b(nestedLoc, nestedBuilder);
+          auto coeff = b.create<tensor::ExtractOp>(coeffs, ValueRange{index});
+          auto cmpOp =
+              b.create<arith::CmpIOp>(arith::CmpIPredicate::sgt, coeff, c0);
```
Comment (asraa) [inline lib/Conversion/PolyToStandard/PolyToStandard.cpp:168]: Small question for me: By using "sgt", do you interpret the unsigned coefficient as signed? Would that be an issue for large coeffs?

It's possible here I think to also use a "not equal to" op.

**Review by asraa — COMMENTED** [review]
Nice! Looks really good. I'll add pull_ready, but just have one question on degree lowering.

File: lib/Conversion/PolyToStandard/PolyToStandard.cpp (hunk 1 above)
Comment (j2kun) [reply lib/Conversion/PolyToStandard/PolyToStandard.cpp:3]: Done

File: tests/poly/lower_monomial.mlir (hunk 2 above)
Comment (j2kun) [reply tests/poly/lower_monomial.mlir:13]: Done.

File: lib/Conversion/PolyToStandard/PolyToStandard.cpp (hunk 3 above)
Comment (j2kun) [reply lib/Conversion/PolyToStandard/PolyToStandard.cpp:168]: Fixed.

**Review by asraa — APPROVED** [review]

**Pull request merged**